### PR TITLE
Move git package from prow-auto-bumper to shared

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Move git package from prow-auto-bumper to shared so that it can be reused.

/cc @chaodaiG

### CODE REVIEW

File: shared/git/info.go [hunk 1]
```diff
@@ -14,19 +14,21 @@ See the License for the specific language governing permissions and
...
-	"log"
-)
+type Info struct {
+    Org      string
+    Repo     string
+    Head     string // PR head branch
+    Base     string // PR base branch
+    UserID   string // Github User ID of PR creator
+    UserName string // User display name for Git commit
+    Email    string // User email address for Git commit
```
Comment (mattmoor) [inline shared/git/info.go]: Format Go code:
```suggestion
	Org      string
	Repo     string
	Head     string // PR head branch
	Base     string // PR base branch
	UserID   string // Github User ID of PR creator
	UserName string // User display name for Git commit
	Email    string // User email address for Git commit
```

File: shared/git/info.go [hunk 2]
```diff
@@ -14,19 +14,21 @@ See the License for the specific language governing permissions and
...
-func run(message string, call func() error, dryrun bool) error {
-	if dryrun {
-		log.Printf("[dry run] %s", message)
-		return nil
-	}
-	log.Printf(message)
-	return call()
+// HeadRef is in the form of "user:head", i.e. "github_user:branch_foo"
+func (gi *Info) GetHeadRef() string {
+    return fmt.Sprintf("%s:%s", gi.UserID, gi.Head)
```
Comment (mattmoor) [inline shared/git/info.go]: Format Go code:
```suggestion
	return fmt.Sprintf("%s:%s", gi.UserID, gi.Head)
```

**Comment by chaodaiG** [comment]
Now that we don't "share" any package with external repo(s), I think we should rename "shared" to "pkg" now

File: pkg/prow/prow.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ import (
 	"strconv"
 	"strings"
 
-	"knative.dev/test-infra/shared/gcs"
+    "knative.dev/test-infra/pkg/gcs"
```
Comment (mattmoor) [inline pkg/prow/prow.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/gcs"
```

File: pkg/testgrid/testgrid.go [hunk 4]
```diff
@@ -24,9 +24,9 @@ import (
 	"os"
 	"path"
 
-	"knative.dev/test-infra/shared/common"
-	"knative.dev/test-infra/shared/junit"
-	"knative.dev/test-infra/shared/prow"
+	"knative.dev/test-infra/pkg/common"
+    "knative.dev/test-infra/pkg/junit"
```
Comment (mattmoor) [inline pkg/testgrid/testgrid.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
```

File: pkg/testgrid/yaml.go [hunk 5]
```diff
@@ -23,7 +23,7 @@ import (
 
 	"gopkg.in/yaml.v2"
 
-	"knative.dev/test-infra/shared/common"
+    "knative.dev/test-infra/pkg/common"
```
Comment (mattmoor) [inline pkg/testgrid/yaml.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/common"
```

File: tools/coverage/testgrid/testsuiteXmlWriter.go [hunk 6]
```diff
@@ -21,7 +21,7 @@ import (
 	"os"
 	"strconv"
 
-	"knative.dev/test-infra/shared/junit"
+    "knative.dev/test-infra/pkg/junit"
```
Comment (mattmoor) [inline tools/coverage/testgrid/testsuiteXmlWriter.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
```

File: tools/flaky-test-reporter/github_issue.go [hunk 7]
```diff
@@ -31,7 +31,7 @@ import (
 
 	"knative.dev/pkg/test/ghutil"
 	"knative.dev/pkg/test/helpers"
-	"knative.dev/test-infra/shared/junit"
+    "knative.dev/test-infra/pkg/junit"
```
Comment (mattmoor) [inline tools/flaky-test-reporter/github_issue.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
```

File: tools/flaky-test-reporter/jsonreport/jsonreport.go [hunk 8]
```diff
@@ -25,8 +25,8 @@ import (
 	"strings"
 	"time"
 
-	"knative.dev/test-infra/shared/common"
-	"knative.dev/test-infra/shared/prow"
+	"knative.dev/test-infra/pkg/common"
+    "knative.dev/test-infra/pkg/prow"
```
Comment (mattmoor) [inline tools/flaky-test-reporter/jsonreport/jsonreport.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/prow"
```

File: tools/flaky-test-reporter/main.go [hunk 9]
```diff
@@ -28,7 +28,7 @@ import (
 
 	"knative.dev/pkg/test/helpers"
 	"knative.dev/pkg/test/slackutil"
-	"knative.dev/test-infra/shared/prow"
+    "knative.dev/test-infra/pkg/prow"
```
Comment (mattmoor) [inline tools/flaky-test-reporter/main.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/prow"
```

File: tools/flaky-test-reporter/result.go [hunk 10]
```diff
@@ -28,9 +28,9 @@ import (
 	"sort"
 	"strings"
 
-	"knative.dev/test-infra/shared/common"
-	"knative.dev/test-infra/shared/junit"
-	"knative.dev/test-infra/shared/prow"
+    "knative.dev/test-infra/pkg/common"
```
Comment (mattmoor) [inline tools/flaky-test-reporter/result.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/common"
```

File: tools/flaky-test-reporter/result.go [hunk 11]
```diff
@@ -28,9 +28,9 @@ import (
...
+	"knative.dev/test-infra/pkg/junit"
+    "knative.dev/test-infra/pkg/prow"
```
Comment (mattmoor) [inline tools/flaky-test-reporter/result.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/prow"
```

File: tools/flaky-test-retryer/log_parser.go [hunk 12]
```diff
@@ -25,8 +25,8 @@ import (
 	"path/filepath"
 	"strings"
 
-	"knative.dev/test-infra/shared/junit"
-	"knative.dev/test-infra/shared/prow"
+    "knative.dev/test-infra/pkg/junit"
```
Comment (mattmoor) [inline tools/flaky-test-retryer/log_parser.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/junit"
```

File: tools/monitoring/alert/alert.go [hunk 13]
```diff
@@ -20,7 +20,7 @@ import (
 	"context"
 	"log"
 
-	"knative.dev/test-infra/shared/gcs"
+    "knative.dev/test-infra/pkg/gcs"
```
Comment (mattmoor) [inline tools/monitoring/alert/alert.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/gcs"
```

File: tools/monitoring/main.go [hunk 14]
```diff
@@ -24,8 +24,8 @@ import (
 	"net/http"
 	"os"
 
-	"knative.dev/test-infra/shared/gcs"
-	"knative.dev/test-infra/shared/mysql"
+    "knative.dev/test-infra/pkg/gcs"
```
Comment (mattmoor) [inline tools/monitoring/main.go]: Format Go code:
```suggestion
	"knative.dev/test-infra/pkg/gcs"
```

**Pull request merged**